### DISCUSSION TITLE
can we reload the excalidraw component manually??

### DISCUSSION THREAD

**Discussion by sohailamjad256** [excalidraw/excalidraw#4458] [Q&A] [answered]

**Comment by dwelle** [comment]
Changing the `key` prop should do it.

**Reply by sohailamjad256** [reply]
i just means to say  like we are saving excalidraw ...so, same like that can we give button and refresh the whole canvas with library items??
if you have answer then can you please share code or screen shot of that code ..it will be helpful

**Reply by ad1992** [reply]
Could you explain the use case, why would you need to refresh the canvas with library items ? Probably you are looking for a way to add new library items to the component programatically ?

**Reply by sohailamjad256** [reply]
yes right..iam just  finding the way to  add some new library items through dropdowns..in the dropdown i will give names of the shapes and whenever we select any shapes names then automatically it need to add in the library items

**Reply by ad1992** [reply]
Ok we will soon be shipping an API to add new library items to Excalidraw in the next version

**Reply by sohailamjad256** [reply]
so..we cant do now??means we cant refresh now excalidraw component
??

**Reply by sohailamjad256** [reply]
can we take a call on this..just for 5 min?

**Reply by ad1992** [reply]
Since we currently don't support adding library programatically but it will be available very soon! 
For a work around what you can do is, you can update `initialData` with new library items and add update `key` prop so `Excalidraw` is force rerendered.

**Reply by sohailamjad256** [reply]
yes i have given one function where iam updating like this..**excalidrawRef.current.updateScene(sceneData)** but its works for only elements and app state but not for library items..i tried already

**Reply by sohailamjad256** [reply]
update scene is a function name..and scene data is object where iam declared elements,library items and app state

**Reply by ad1992** [reply]
Nope I didn't mean `updateScene`. 
Please try by updating the `initialData` instead and update key prop as mentioned below, you will need to make sure that `initialData` has all latest data before rerendering so you don't lose any data
> Since we currently don't support adding library programatically but it will be available very soon! For a work around what you can do is, you can update `initialData` with new library items and add update `key` prop so `Excalidraw` is force rerendered.

**Reply by sohailamjad256** [reply]
can you show some code ..how i can use that update key prop..

**Comment by ad1992** [comment, marked as answer]
You can update the `initialData` with updated data which contains new library items as well and once thats done you will need to update the `key` attribute which will trigger the re render automatically, added an [code sandbox example](https://1m7pt.csb.app/) for the same, try clicking on "Force update" button and see the library items and elements are getting updated since key is updated 

```
<Excalidraw key={your key} />
```
![React App](https://user-images.githubusercontent.com/11256141/147064533-eb2eb067-7a0d-4074-88f3-b8e0479402ba.gif)

**Reply by sohailamjad256** [reply]
yes..i think this will help me

**Reply by sohailamjad256** [reply]
Hi...Thanks 
Now force update is working .but if iam clicking force update  button without saving the existing data ..that time i am loosing existing data

**Reply by ad1992** [reply]
Yes as mentioned earlier, you need to make sure that latest data is sent to Excalidraw when force updating so there is no data loss as Excalidraw will be rerendered with `initialData`
